lib/rouge/lexers/liquid.rb

```diff
@@ -59,13 +59,13 @@ class Liquid < RegexLexer
         rule %r/include/, Name::Tag, :include
 
         # end of block
-        rule %r/(end(case|unless|if))(\s*)(%\})/ do
-          groups Keyword::Reserved, nil, Text::Whitespace, Punctuation
+        rule %r/(end(?:case|unless|if))(\s*)(%\})/ do
```

Comment:
`{% for ... in ... %} ... {% endfor %}` is another builtin block tag.
I've not checked the entire lexer, so not sure if it is handled as well..

Reply:
While I think we should definitely make that change, I think it's sufficiently different that it should be its own PR. I'm going to file an issue about it with one of our fancy new templates.

Reply:
Added it as #1194.